fog/ledger/server/src/db_fetcher.rs

```diff
@@ -234,59 +237,60 @@ impl<
                     e
                 );
                 std::thread::sleep(Self::ERROR_RETRY_FREQUENCY);
-                *next_block_index
+                // We errored so we don't know how many blocks are in the ledger
+                return 0;
```

Comment:
returning zero here could still cause us to be in the same situation where 0 is returned for all of the stores, but this is an error case that would only happen if we have some other issue so it seems tolerable to do here, we could also fall back to `shared_state.highest_processed_block_count`